### PR TITLE
table::column_table should be storage::column_table

### LINKED ISSUE #118: core/column/table should move into core/storage
table is a kind of storage!

### PR DESCRIPTION
**Pull request by kumagi**
fix #118 
move from `jubatus/core/table/column/*` to `jubatus/core/storage/`.
and using `column_table`'s `bit_vector` implementation globally.

It also includes some refactoring for using forward declaration instead of `include`.

### CODE REVIEW

File: jubatus/core/bandit/bandit_base.hpp [hunk 1]
```diff
@@ -20,11 +20,21 @@
 #include <string>
 
 #include "arm_info.hpp"
-#include "../framework/packer.hpp"
 #include "../common/version.hpp"
```
Comment (gintenlabo) [inline jubatus/core/bandit/bandit_base.hpp]: This can be removed.

File: jubatus/core/bandit/epsilon_greedy.hpp [hunk 2]
```diff
@@ -25,6 +25,9 @@
 
 namespace jubatus {
 namespace core {
+namespace storage {
+class version;
```
Comment (gintenlabo) [inline jubatus/core/bandit/epsilon_greedy.hpp]: This can be removed because `bandit_base.hpp` declares it.

File: jubatus/core/bandit/exp3.hpp [hunk 3]
```diff
@@ -26,6 +26,9 @@
 
 namespace jubatus {
 namespace core {
+namespace storage {
+class version;
```
Comment (gintenlabo) [inline jubatus/core/bandit/exp3.hpp]: ditto.

File: jubatus/core/bandit/softmax.hpp [hunk 4]
```diff
@@ -19,12 +19,15 @@
...
 
+#include "jubatus/util/math/random.h"
 #include "bandit_base.hpp"
 #include "summation_storage.hpp"
-#include "jubatus/util/math/random.h"
 
 namespace jubatus {
 namespace core {
+namespace storage {
+class version;
```
Comment (gintenlabo) [inline jubatus/core/bandit/softmax.hpp]: ditto.

File: jubatus/core/bandit/ucb1.hpp [hunk 5]
```diff
@@ -24,6 +24,9 @@
 
 namespace jubatus {
 namespace core {
+namespace storage {
+class version;
```
Comment (gintenlabo) [inline jubatus/core/bandit/ucb1.hpp]: ditto.

File: jubatus/core/recommender/euclid_lsh.hpp [hunk 6]
```diff
@@ -25,10 +25,21 @@
...
 namespace core {
+namespace framework {
+template <typename Model, typename Diff>
+class linear_mixable_helper;
+}  // namespace framework
+namespace storage {
+class lsh_index_storage;
+struct lsh_entry;
+typedef jubatus::util::data::unordered_map<std::string, lsh_entry>
+  lsh_master_table_t;
```
Comment (gintenlabo) [inline jubatus/core/recommender/euclid_lsh.hpp]: wrong indent

File: jubatus/core/bandit/bandit_base.hpp (hunk 1 above)
Comment (kumagi) [inline jubatus/core/bandit/bandit_base.hpp]: Yes, I can be removed. Thank you, I'll fix it

File: jubatus/core/bandit/epsilon_greedy.hpp (hunk 2 above)
Comment (kumagi) [inline jubatus/core/bandit/epsilon_greedy.hpp]: thanks.

File: jubatus/core/bandit/exp3.hpp (hunk 3 above)
Comment (kumagi) [inline jubatus/core/bandit/exp3.hpp]: ok

File: jubatus/core/bandit/softmax.hpp (hunk 4 above)
Comment (kumagi) [inline jubatus/core/bandit/softmax.hpp]: thanks.

File: jubatus/core/bandit/ucb1.hpp (hunk 5 above)
Comment (kumagi) [inline jubatus/core/bandit/ucb1.hpp]: thanks.

File: jubatus/core/recommender/euclid_lsh.hpp (hunk 6 above)
Comment (kumagi) [inline jubatus/core/recommender/euclid_lsh.hpp]: I'll fix it.

**Comment by gintenlabo** [comment]
:+1: LGTM

**Comment by kumagi** [comment]
Thank you.

**Pull request merged**